### DISCUSSION TITLE
Caught plotting error: Matches do not match with number of write entries

### DISCUSSION THREAD

**Discussion by cccat6** [Chia-Network/chia-blockchain#2742] [Q&A] [answered]
Any idea of why that happened?
It is 3900x + 2*2T NVMe SSD + 128G RAM. There are enough spaces for temp cache. All the plot tasks on both drives show the same problem, It could appear in any table of phase 1. Every attempt fail in this way.

```
	Bucket 123 uniform sort. Ram: 7.934GiB, u_sort min: 2.750GiB, qs min: 0.688GiB.
	Bucket 124 uniform sort. Ram: 7.934GiB, u_sort min: 2.750GiB, qs min: 0.688GiB.
	Bucket 125 uniform sort. Ram: 7.934GiB, u_sort min: 1.375GiB, qs min: 0.687GiB.
	Bucket 126 uniform sort. Ram: 7.934GiB, u_sort min: 2.750GiB, qs min: 0.688GiB.
	Bucket 127 uniform sort. Ram: 7.934GiB, u_sort min: 2.750GiB, qs min: 0.688GiB.
	Total matches: 4293773604
Caught plotting error: Matches do not match with number of write entries 4293773604 4293768975
Traceback (most recent call last):
  File "/usr/local/bin/chia", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/chia/cmds/chia.py", line 77, in main
    cli()  # pylint: disable=no-value-for-parameter
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/dist-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/chia/cmds/plots.py", line 135, in create_cmd
    create_plots(Params(), ctx.obj["root_path"])
  File "/usr/local/lib/python3.7/dist-packages/chia/plotting/create_plots.py", line 176, in create_plots
    args.nobitfield,
RuntimeError: std::exception
```

**Comment by adamszen** [comment]
I have the same error. Machine for plotting are server with ECC RAM. No other logs, hardware is OK. 

> Bucket 126 uniform sort. Ram: 9.706GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
> Bucket 127 uniform sort. Ram: 9.706GiB, u_sort min: 1.125GiB, qs min: 0.281GiB.
> Total matches: 4064710477
> **Caught plotting error: Matches do not match with number of write entries 4064710477 4064356615**
> Traceback (most recent call last):
>   File "/root/chia-blockchain/venv/bin/chia", line 33, in <module>
>     sys.exit(load_entry_point('chia-blockchain', 'console_scripts', 'chia')())
>   File "/root/chia-blockchain/chia/cmds/chia.py", line 77, in main
>     cli()  # pylint: disable=no-value-for-parameter
>   File "/root/chia-blockchain/venv/lib/python3.7/site-packages/click/core.py", line 829, in __call__
>     return self.main(*args, **kwargs)
>   File "/root/chia-blockchain/venv/lib/python3.7/site-packages/click/core.py", line 782, in main
>     rv = self.invoke(ctx)
>   File "/root/chia-blockchain/venv/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
>     return _process_result(sub_ctx.command.invoke(sub_ctx))
>   File "/root/chia-blockchain/venv/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
>     return _process_result(sub_ctx.command.invoke(sub_ctx))
>   File "/root/chia-blockchain/venv/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/root/chia-blockchain/venv/lib/python3.7/site-packages/click/core.py", line 610, in invoke
>     return callback(*args, **kwargs)
>   File "/root/chia-blockchain/venv/lib/python3.7/site-packages/click/decorators.py", line 21, in new_func
>     return f(get_current_context(), *args, **kwargs)
>   File "/root/chia-blockchain/chia/cmds/plots.py", line 135, in create_cmd
>     create_plots(Params(), ctx.obj["root_path"])
>   File "/root/chia-blockchain/chia/plotting/create_plots.py", line 176, in create_plots
>     args.nobitfield,
> RuntimeError: std::exception

Heellp! :)

**Comment by adamszen** [comment]
> Any idea of why that happened?

@cccat6 what linux distriubution and Python version You are using?

**Reply by cccat6** [reply]
I tried Ubuntu20 + Python3.8 and Debian10 + Python3.7. Both get the same problem.
I am currently switching to Windows and see what is going to be.

**Reply by adamszen** [reply]
I testing on Debian10 + Python3.7 (3.7.4 originally and 3.7.9 compiled from source) - the same issue. Sometimes exit with other error:

> Bucket 0 uniform sort. Ram: 9.699GiB, u_sort min: 2.250GiB, qs min: 0.563GiB.
> terminate called after throwing an instance of 'InvalidStateException'
> terminate called recursively
> what():  Offset too large: 37060
> Aborted

It's a bit frustrating - i can't create plots :(

**Reply by adamszen** [reply]
> I tried Ubuntu20 + Python3.8 and Debian10 + Python3.7. Both get the same problem.
> I am currently switching to Windows and see what is going to be.

What filesystem are you using for the temporary directory? If i use RAID0 for temporary directory from ~8 disks and ext4 - it will fail with these errors. If i using single disks (without raid) for temporary directory - i think it works good (for now successfully completed three phases with five plots simultaneously).

**Reply by cccat6** [reply]
ext4, not raid

**Comment by cccat6** [comment, marked as answer]
I have solved this problem with change the OS to Windows rather than Linux. I didn't find the actual problem and solution, but I waste too much time on it.
I don't really know what makes Windows work but Linux not. Maybe based on the file system, or maybe the problem of kernel like memory management.
It can plot now and that's fine. Although Windows is slower than Linux, at least it works.

**Comment by lazydrone** [comment]
I think I'm going to do the same (switch to Win10).  I have four machines farming, two with Linux Ubuntu and two Windows.  Have never had a problem with the windows machines, although admittedly, I haven't really pushed them.  And one of them is a 6 GPU mining rig (doing about 2 plots at a time in the background).  No issues farming or mining.
The linux machine that's giving me the most problems is also the most powerful one.  Threadripper 3970 with 192g ram, (2) 3.8TB enterprise U.2 drives plus a 1/2 dozen 18 TB HDDs.  I've tried a (clean) install.  I've tried the GUI, the Command Line.  1 plot drive, 2 plot drives. In three days I've only managed 14 plots, where my "part time" windows machines have done about 40 combined.  So frustrating to see multiple plots fail 4 hours into it.  And it's not happening in phase 1 which is supposed to be the most resource intensive.   In real farming it's always the weather or the bugs that get you....or maybe I'm just a poor farmer.

I'm glad I saw your post, because I was just thinking that this morning (to change the OS).  I'm not really a linux person anyway.  It's ok, but it's like using your mouse in your left hand, you can do it, but you don't like it and you're inefficient as hell.
